### DISCUSSION TITLE
Why copy-paste?

### DISCUSSION THREAD

**Discussion by salomvary** [gluestack/gluestack-ui#2436] [General]
Hi folks,

I'm new to Gluestack UI and not been doing front-end development in the last few months and missed out on the most recent trends ;) Was wondering why Gluestack UI promotes copypasting as opposed to old-school dependency management. Tried looking into the Introduction and FAQ sections, to no avail.

Someone please enlighten me :)

Thanks!

**Comment by sanketsahu** [comment]
Hi @salomvary, after building NativeBase and gluestack-ui v1, we realised that the right abstraction for component libraries is to ship it without styles and let the user manage the styles as per their needs. So, we ship unstyled components like the @gluestack-ui/button, which provides the right API and structure, and the styling is copied into the project, which uses these unstyled components.

This approach is similar to shadcn/ui, which uses Radix UI under the hood and styles with Tailwind on top of it, which is copied in user's projects.
